# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/script/ScriptLoader.h
# Commit: c11cffaded9c
# Full Hash: c11cffaded9cfed53b04e5c7875278e9a9e55396
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r 0527d227c6b6 -r c11cffaded9c dom/script/ScriptLoader.h
--- a/dom/script/ScriptLoader.h	Fri Feb 18 12:45:52 2022 +0000
+++ b/dom/script/ScriptLoader.h	Fri Feb 18 12:45:53 2022 +0000
@@ -8,6 +8,9 @@
 #define mozilla_dom_ScriptLoader_h
 
 #include "js/TypeDecls.h"
+#include "js/loader/LoadedScript.h"
+#include "js/loader/ScriptKind.h"
+#include "js/loader/ScriptLoadRequest.h"
 #include "nsCOMPtr.h"
 #include "nsRefPtrHashtable.h"
 #include "nsIScriptElement.h"
@@ -20,13 +23,10 @@
 #include "nsIScriptLoaderObserver.h"
 #include "nsURIHashKey.h"
 #include "mozilla/CORSMode.h"
-#include "mozilla/dom/LoadedScript.h"
 #include "mozilla/dom/JSExecutionContext.h"  // JSExecutionContext
-#include "mozilla/dom/ScriptLoadRequest.h"
 #include "ModuleLoader.h"
 #include "mozilla/MaybeOneOf.h"
 #include "mozilla/MozPromise.h"
-#include "ScriptKind.h"
 
 class nsCycleCollectionTraversalCallback;
 class nsIChannel;
@@ -44,6 +44,16 @@
 template <typename UnitT>
 class SourceText;
 
+namespace loader {
+
+class LoadedScript;
+class ScriptLoaderInterface;
+class ModuleLoadRequest;
+class ModuleScript;
+class ScriptLoadRequest;
+class ScriptLoadRequestList;
+
+}  // namespace loader
 }  // namespace JS
 
 namespace mozilla {
@@ -56,18 +66,11 @@
 class AutoJSAPI;
 class DocGroup;
 class Document;
-class LoadedScript;
-class ScriptLoaderInterface;
 class ModuleLoader;
-class ModuleLoadRequest;
-class ModuleScript;
 class SRICheckDataVerifier;
 class SRIMetadata;
 class ScriptLoadHandler;
-class ScriptLoadRequest;
-class ScriptLoadRequestList;
 class ScriptLoadContext;
-
 class ScriptLoader;
 class ScriptRequestProcessor;
 
@@ -96,7 +99,7 @@
 // Script loader implementation
 //////////////////////////////////////////////////////////////
 
-class ScriptLoader final : public ScriptLoaderInterface {
+class ScriptLoader final : public JS::loader::ScriptLoaderInterface {
   class MOZ_STACK_CLASS AutoCurrentScriptUpdater {
    public:
     AutoCurrentScriptUpdater(ScriptLoader* aScriptLoader,
@@ -117,7 +120,7 @@
     ScriptLoader* mScriptLoader;
   };
 
-  friend class ModuleLoadRequest;
+  friend class JS::loader::ModuleLoadRequest;
   friend class ScriptRequestProcessor;
   friend class ModuleLoader;
   friend class ScriptLoadHandler;
@@ -409,7 +412,7 @@
    * Get the currently active script. This is used as the initiating script when
    * executing timeout handler scripts.
    */
-  static LoadedScript* GetActiveScript(JSContext* aCx);
+  static JS::loader::LoadedScript* GetActiveScript(JSContext* aCx);
 
   Document* GetDocument() const { return mDocument; }
 
@@ -439,9 +442,9 @@
 
   bool ProcessInlineScript(nsIScriptElement* aElement, ScriptKind aScriptKind);
 
-  ScriptLoadRequest* LookupPreloadRequest(nsIScriptElement* aElement,
-                                          ScriptKind aScriptKind,
-                                          const SRIMetadata& aSRIMetadata);
+  JS::loader::ScriptLoadRequest* LookupPreloadRequest(
+      nsIScriptElement* aElement, ScriptKind aScriptKind,
+      const SRIMetadata& aSRIMetadata);
 
   void GetSRIMetadata(const nsAString& aIntegrityAttr,
                       SRIMetadata* aMetadataOut);